pkg/apis/networking/v1/types.go

```diff
@@ -174,3 +175,149 @@ type NetworkList struct {
 
 	Items []Network `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// RemoteCluster is a specification for a RemoteCluster resource
+type RemoteCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RemoteClusterSpec   `json:"spec"`
+	Status RemoteClusterStatus `json:"status"`
+}
+
+// RemoteClusterSpec is the spec for a RemoteCluster resource
+type RemoteClusterSpec struct {
+	// globally unique id
+	ClusterID uint32 `json:"cluster_id"`
+	// just for human memory, no practical use
+	ClusterName string              `json:"cluster_name"`
```

Comment:
metav1.TypeMeta already has a 'Name' filed

Reply:
yes, clustername is redundant. I prefer deleting it.